symengine/cwrapper.cpp

```diff
@@ -58,6 +58,25 @@ inline bool is_aligned(T *p, size_t n = alignof(T))
 
 extern "C" {
 
+
+#define CWRAPPER_BEGIN()                                                       \
+    try                                                                        \
+    {
+
+#define CWRAPPER_END(error, code)                                              \
+        return 0;                                                              \
+    }                                                                          \
+    catch ( error )                                                            \
+    {                                                                          \
+        return code;                                                           \
+    }
+
+#define CWRAPPER_END_EXTRA(error, code)                                        \
```

Comment:
@isuruf yes.. I overlooked that. So do you suggest that I add catch statements for all the exceptions in the `CWRAPPER_END` macro, and get rid of the `CWRAPPER_END_EXTRA` macro?